test/es-module/test-esm-snapshot.mjs

```diff
@@ -1,7 +0,0 @@
-// Flags: --experimental-modules
-/* eslint-disable required-modules */
-import './esm-snapshot-mutator';
-import one from './esm-snapshot';
-import assert from 'assert';
-
-assert.strictEqual(one, 1);
```

Comment:
this file is done to test timing of snapshotting. It should not be removed

Reply:
so this test is checking if mutating require.cache affects the imported modules? Seeing as module._load is used wouldn't this be the expected behavior?

Reply:
> wouldn't this be the expected behavior?

I don't know what "this" is.

Reply:
sorry about that, i meant wouldn't
1) importing the mutator
2) importing the mutatee (i hope thats a word)
3) mutatee being mutated

be the expected behavior

Reply:
If we walk thought the code the following happens:

* `require('./esm-snapshot-mutator')`
    * Module Record is formed for `./esm-snapshot-mutator`
    * `require('./esm-snapshot')`
        * Module Record is formed for `./esm-snapshot`
        * Evaluation of `./esm-snapshot` ends, `module.exports === 1`
        * Snapshot takes effect, Module Record saves as `default = 1`
     * Evaluation of `./esm-snapshot-mutator` ends

The snapshot is taken before [`./esm-snapshot-mutator` changes the entry of `require.cache](https://github.com/nodejs/node/blob/c8a389e19f172edbada83f59944cad7cc802d9d5/test/es-module/esm-snapshot-mutator.js#L5).

This snapshot ensures the Module Namespace values of `./esm-snapshot` is the same no matter when it is imported.